pkg/controller/usbdevice/usbdevice_controller.go

```diff
@@ -0,0 +1,217 @@
+package usbdevice
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/wrangler/pkg/relatedresource"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/harvester/pcidevices/pkg/apis/devices.harvesterhci.io/v1beta1"
+	"github.com/harvester/pcidevices/pkg/deviceplugins"
+	ctldevicerv1vbeta1 "github.com/harvester/pcidevices/pkg/generated/controllers/devices.harvesterhci.io/v1beta1"
+	"github.com/harvester/pcidevices/pkg/util/gousb"
+	"github.com/harvester/pcidevices/pkg/util/gousb/usbid"
+)
+
+type Handler struct {
```

Comment:
we should pass the cache interfaces for both types, and change all get/lists to use the cache